### PR TITLE
Update NNPI Backend to v0.5.1.4

### PR DESCRIPTION
**Pull request by omromano**
Summary: Update of NNPI Backend to v0.5.1.4 branched out of commit 2aa0e5d3e108a0607acd3184dc803cfd77bd6c3c.

### CODE REVIEW

**Comment by omromano** [comment]
CC: @jsubag

File: lib/Backends/NNPI/NNPIOptions.h [hunk 1]
```diff
@@ -300,17 +325,32 @@ class NNPIDeviceOptions : public NNPIOptions {
   /// Dump IO to files.
   DECLARE_NNPI_OPTION(dumpIOtoFiles, bool, "DumpIOtoFiles",
                       "Dump Inputs/Outputs to files.", "NNPI_DUMP_IO", "0");
+  /// Force using a specific AVX type.
+  DECLARE_NNPI_OPTION(AvxType, int, "AvxType",
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPIOptions.h]: nit: `avxType` to match lowerCamelCase for the other options?

File: lib/Backends/NNPI/NNPIResource.h [hunk 2]
```diff
@@ -92,12 +93,13 @@ class NNPIResource {
...
   ResourceUsage usage_;
-  const NNPIDeviceOptions *deviceOptions_;
+  std::shared_ptr<NNPIDeviceOptions> deviceOptions_;
   std::vector<uint8_t> refStorage_;
   uint32_t cmdListIdx_;
 
   /// Update the owned host resource with data taken from the given tensor.
-  void UpdateHostResourceFromTensor(Tensor *t, bool partialTensor);
+  // return true when successfull, false otherwise.
+  bool UpdateHostResourceFromTensor(Tensor *t, bool partialTensor);
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPIResource.h]: nit: lowercase first word (e.g. `updateHost...`

File: lib/Backends/NNPI/NNPIResource.h [hunk 3]
```diff
@@ -50,7 +50,8 @@ class NNPIResource {
                                       std::function<void(Error)> resultCB);
 
   /// Initialize a resource.
-  bool Init(const NNPIObjectName name, const NNPIDeviceOptions *deviceOptions,
+  bool Init(const NNPIObjectName name,
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPIResource.h]: nit: lowercase `init`.

File: lib/Backends/NNPI/NNPIUtils.h [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+#ifndef GLOW_BACKENDS_NNPI_NNPIUTILS_H
+#define GLOW_BACKENDS_NNPI_NNPIUTILS_H
+
+#include <cstdint>
+
+using namespace std;
+
+enum NNPIAVXType { NNPI_AVX_NONE = 0, NNPI_AVX_AVX512 };
+
+inline void ConvertI64toI32(int64_t const *i64Data, int32_t *i32Data,
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPIUtils.h]: nit: lowercase `convertI64...`

File: lib/Backends/NNPI/NNPIUtils.h [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+                            uint32_t elements) {
+  for (size_t i = 0; i < elements; i++) {
+    i32Data[i] = static_cast<int32_t>(i64Data[i]);
+  }
+}
+void ConvertI64toI32_AVX512(int64_t const *i64Data, int32_t *i32Data,
```
Comment (jfix71) [inline lib/Backends/NNPI/NNPIUtils.h]: nit: lowercase `convertI64...`

**Pull request closed without merging**